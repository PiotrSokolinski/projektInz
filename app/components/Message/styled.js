import styled from 'styled-components'
import { Colors } from 'themes'
export const Container = styled.div`
  display: flex;
  margin: 15px 5px;
  flex-direction: ${props => (props.isOwner ? 'row-reverse' : 'row')};
  align-items: flex-end;
`

export const MessageCloud = styled.div`
  background-color: ${props => (props.isOwner ? Colors.anakiwa : Colors.athensGrayLight)};
  max-width: 75%;
  border-radius: 10px;
  padding: 10px;
  margin: 0 5px;
  overflow-wrap: break-word;
  ${props => (props.isOwner ? 'border-bottom-right-radius: 0;' : 'border-bottom-left-radius: 0;')}
`
export const MessageTime = styled.div`
  font-size: 13px;
`
