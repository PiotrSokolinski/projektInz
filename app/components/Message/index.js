/**
 *
 * Message
 *
 */

import React from 'react'
import PropTypes from 'prop-types'

import UserAvatar from 'components/UserAvatar'

import * as Styled from './styled'

const Message = ({ text, isOwner, time }) => (
  <Styled.Container isOwner={isOwner}>
    <UserAvatar size="tiny" />
    <Styled.MessageCloud isOwner={isOwner}>{text}</Styled.MessageCloud>
    <Styled.MessageTime>{time}</Styled.MessageTime>
  </Styled.Container>
)

Message.propTypes = {
  text: PropTypes.string,
  isOwner: PropTypes.bool,
  time: PropTypes.string,
}

export default Message
