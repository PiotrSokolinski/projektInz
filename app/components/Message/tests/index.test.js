/**
 *
 * Tests for Message
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'

import Message from '../index'

describe('<Message />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(<Message />)
    expect(spy).not.toHaveBeenCalled()
  })

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Message />)
    expect(firstChild).toMatchSnapshot()
  })
})
