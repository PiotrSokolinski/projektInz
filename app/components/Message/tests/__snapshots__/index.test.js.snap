// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> Should render and match the snapshot 1`] = `
<div
  class="styled__Container-lr2zll-0 hiCLbP"
>
  <div
    class="tiny styled__Container-vinm4t-0 lmbuMq"
  >
    <img
      alt="avatar"
      class="avatar-image"
      src="IMAGE_MOCK"
    />
  </div>
  <div
    class="styled__MessageCloud-lr2zll-1 dxMckx"
  />
  <div
    class="styled__MessageTime-lr2zll-2 gTomPT"
  />
</div>
`;
